options/linux/generic/sys-prctl-stubs.cpp

```diff
@@ -1,12 +1,32 @@
 
+#include <stdarg.h>
+#include <errno.h>
 #include <bits/ensure.h>
 #include <sys/prctl.h>
 
 #include <mlibc/debug.hpp>
 
-int prctl(int, ...) {
-	mlibc::infoLogger() << "\e[31mmlibc: prctl() is not implemented correctly\e[39m"
-			<< frg::endlog;
+#include "mlibc/linux-sysdeps.hpp"
+
+int prctl(int op, ...) {
+	if(!mlibc::sys_prctl) {
+		MLIBC_MISSING_SYSDEP();
+		errno = ENOSYS;
+		return -1;
+	}
+
+	unsigned long x[4];
+	va_list ap;
+	va_start(ap, op);
+	for(int i = 0; i < 4; i++)
+		x[i] = va_arg(ap, unsigned long);
+	va_end(ap);
+	int val;
+
+	if(int e = mlibc::sys_prctl(op, x[0], x[1], x[2], x[3], &val); e) {
```

Comment:
I dislike this interface to sys_prctl. What if Linux decides to add another argument? We could also pass the va_list here.

Reply:
We could pass a view type, for something more C++ friendly.

Reply:
Is Linux changing syscall args really a concern? I'd guess they would provide a new, numbered syscall instead.

Reply:
It probably is a concern, since musl has made it a variadic function.